tests/integ/test_stored_procedure_profiler.py

```diff
@@ -0,0 +1,150 @@
+):
+    @sproc(name="table_sp", replace=True)
+    def table_sp(session: snowflake.snowpark.Session) -> DataFrame:
+        return session.sql("select 1")
+
+    pro = profiler_session.stored_procedure_profiler
+    pro.register_modules(["table_sp"])
+    pro.set_targeted_stage(
+        f"{db_parameters['database']}.{db_parameters['schema']}.{tmp_stage_name}"
+    )
+
+    pro.set_active_profiler("LINE")
+
+    profiler_session.call("table_sp")
+    res = pro.get_output()
+    pro.disable()
+
+    pro.register_modules([])
+    assert res is not None
+    assert "Modules Profiled" in res
+
+
+@pytest.mark.skipif(
+    "config.getoption('local_testing_mode', default=False)",
+    reason="session.sql is not supported in localtesting",
+)
+def test_single_return_value_of_sp(
+    is_profiler_function_exist, profiler_session, db_parameters, tmp_stage_name
+):
+    @sproc(name="single_value_sp", replace=True)
+    def single_value_sp(session: snowflake.snowpark.Session) -> str:
+        return "success"
+
+    profiler_session.stored_procedure_profiler.register_modules(["single_value_sp"])
+    profiler_session.stored_procedure_profiler.set_targeted_stage(
+        f"{db_parameters['database']}.{db_parameters['schema']}.{tmp_stage_name}"
+    )
+
+    profiler_session.stored_procedure_profiler.set_active_profiler("LINE")
+
+    profiler_session.call("single_value_sp")
+    res = profiler_session.stored_procedure_profiler.get_output()
+
+    profiler_session.stored_procedure_profiler.disable()
+
+    profiler_session.stored_procedure_profiler.register_modules([])
+    assert res is not None
+    assert "Modules Profiled" in res
+
+
+@pytest.mark.skipif(
+    "config.getoption('local_testing_mode', default=False)",
+    reason="session.sql is not supported in localtesting",
+)
+def test_anonymous_procedure(
+    is_profiler_function_exist, profiler_session, db_parameters, tmp_stage_name
+):
+    def single_value_sp(session: snowflake.snowpark.Session) -> str:
+        return "success"
+
+    single_value_sp = profiler_session.sproc.register(single_value_sp, anonymous=True)
+
+    profiler_session.stored_procedure_profiler.set_targeted_stage(
+        f"{db_parameters['database']}.{db_parameters['schema']}.{tmp_stage_name}"
+    )
+
+    profiler_session.stored_procedure_profiler.set_active_profiler("LINE")
+
+    single_value_sp()
+    res = profiler_session.stored_procedure_profiler.get_output()
+
+    profiler_session.stored_procedure_profiler.disable()
+
+    profiler_session.stored_procedure_profiler.register_modules([])
+    assert res is not None
+    assert "Modules Profiled" in res
+
+
+def test_set_incorrect_active_profiler(profiler_session):
+    with pytest.raises(ValueError) as e:
+        profiler_session.stored_procedure_profiler.set_active_profiler(
+            "wrong_active_profiler"
+        )
+    assert "active_profiler expect 'LINE', 'MEMORY'" in str(e)
+
+
+def test_sp_call_match(profiler_session):
```

Comment:
add test cases for `"CALL MY_SPROC()"`, `"    CALL MY_SPROC()"`, `"WITH my procedure"`, `"   WITH my procedure..."`, `"WITH SNOWPARK_TEMP_CTE_1234 AS (SELECT 1 as A) SELECT * FROM SNOWPARK_TEMP_CTE_1234"`

Reply:
Added, but I changed some because anonymous sp call has to be `WITH ... AS PROCEDURE ... CALL ...`

Reply:
would be better if we parametrize this input.